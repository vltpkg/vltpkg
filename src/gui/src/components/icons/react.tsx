export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
