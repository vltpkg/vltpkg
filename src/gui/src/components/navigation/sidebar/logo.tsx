import {
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
} from '@/components/ui/sidebar.jsx'
import { VLTV } from '@/components/icons/vlt-v.jsx'
import { useTheme } from '@/components/ui/theme-provider.jsx'

const SidebarLogo = () => {
  const { resolvedTheme: theme } = useTheme()

  return (
    <SidebarMenu>
      <SidebarMenuItem>
        <SidebarMenuButton
          size="lg"
          className="whitespace-nowrap data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground"
          asChild>
          <a href="/">
            <div className="flex aspect-square h-full items-center justify-center">
              <VLTV
                color={theme === 'dark' ? 'white' : 'black'}
                className="flex items-center justify-center"
              />
            </div>
            <div className="grid flex-1 text-left text-sm leading-tight">
              <span className="truncate font-semibold">
                vlt
                <span className="ml-1 truncate font-light">
                  /volt/
                </span>
              </span>
            </div>
          </a>
        </SidebarMenuButton>
      </SidebarMenuItem>
    </SidebarMenu>
  )
}

export { SidebarLogo }
