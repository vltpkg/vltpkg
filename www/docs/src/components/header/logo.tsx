import { useStore } from '@/state'

const Logo = () => {
  const { getResolvedTheme } = useStore()
  const theme = getResolvedTheme()

  return (
    <img
      src={
        theme === 'dark' ?
          '/logos/vlt-logo-light.png'
        : '/logos/vlt-logo-dark.png'
      }
      height={43}
      width={94}
    />
  )
}

export default Logo
