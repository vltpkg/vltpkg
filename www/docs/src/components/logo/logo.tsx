const Logo = () => {
  return (
    <a href="/" className="relative">
      <svg width="0" height="0">
        <clipPath id="clipPathId">
          <path d="M52 52C47.84 52 43.85 50.35 40.89 47.5H39.12C36.16 50.35 32.17 52 28.01 52C19.19 52 12.01 44.82 12.01 36C12.01 34.77 12.15 33.56 12.42 32.38L11.87 31.47C4.94 29.64 0 23.41 0 16C0 7.18 7.18 0 16 0C20.77 0 25.07 2.1 28 5.43C30.93 2.1 35.23 0 40 0C44.16 0 48.15 1.65 51.11 4.5H52.88C55.84 1.65 59.83 0 63.99 0C68.15 0 72.14 1.65 75.1 4.5H76.87C79.83 1.65 83.82 0 87.98 0C96.8 0 103.98 7.18 103.98 16C103.98 24.82 96.8 32 87.98 32C83.82 32 79.83 30.35 76.87 27.5H75.1C73.15 29.39 70.74 30.75 68.12 31.45L67.57 32.37C67.84 33.55 67.98 34.76 67.98 35.99C67.98 44.81 60.8 51.99 51.98 51.99L52 52Z" />
        </clipPath>
      </svg>
      <svg
        width="94"
        height="43"
        viewBox="0 0 94 43"
        fill="none"
        className="absolute left-[5px] top-1 z-[3] text-black dark:text-white">
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M74.14 5.05267C76.2 2.25267 79.5 0.552673 83.01 0.552673C89.08 0.552673 94.01 5.48267 94.01 11.5527C94.01 17.6227 89.08 22.5527 83.01 22.5527C79.49 22.5527 76.2 20.8527 74.14 18.0527H67.88C66.02 20.5827 63.15 22.2127 60.02 22.5027L57.15 27.2827C57.72 28.6327 58.01 30.0827 58.01 31.5527C58.01 37.6227 53.08 42.5527 47.01 42.5527C43.49 42.5527 40.2 40.8527 38.14 38.0527H31.88C29.82 40.8527 26.52 42.5527 23.01 42.5527C16.94 42.5527 12.01 37.6227 12.01 31.5527C12.01 30.0827 12.31 28.6227 12.87 27.2827L10 22.5027C4.41001 21.9927 0.0100098 17.2827 0.0100098 11.5527C0.0100098 5.48267 4.94001 0.552673 11.01 0.552673C17.08 0.552673 22.01 5.48267 22.01 11.5527C22.01 13.0227 21.71 14.4827 21.15 15.8227L23.01 18.9227L24.87 15.8227C24.3 14.4727 24.01 13.0227 24.01 11.5527C24.01 5.48267 28.94 0.552673 35.01 0.552673C38.53 0.552673 41.82 2.25267 43.88 5.05267H50.14C52.2 2.25267 55.49 0.552673 59.01 0.552673C62.53 0.552673 65.82 2.25267 67.88 5.05267H74.14ZM77.21 13.0527C77.88 15.6327 80.22 17.5527 83.01 17.5527C86.32 17.5527 89.01 14.8627 89.01 11.5527C89.01 8.24267 86.32 5.55267 83.01 5.55267C80.22 5.55267 77.88 7.47267 77.21 10.0527H64.82C64.15 7.47267 61.81 5.55267 59.02 5.55267C56.23 5.55267 53.89 7.47267 53.22 10.0527H40.83C40.16 7.47267 37.82 5.55267 35.03 5.55267C31.72 5.55267 29.03 8.24267 29.03 11.5527C29.03 13.1927 29.69 14.6827 30.76 15.7627L24.73 25.8027C24.19 25.6427 23.62 25.5527 23.03 25.5527C22.44 25.5527 21.87 25.6427 21.33 25.8027L15.3 15.7627C16.37 14.6727 17.03 13.1927 17.03 11.5527C17.03 8.24267 14.34 5.55267 11.03 5.55267C7.72001 5.55267 5.03001 8.24267 5.03001 11.5527C5.03001 14.8627 7.72001 17.5527 11.03 17.5527C11.62 17.5527 12.19 17.4627 12.73 17.3027L18.76 27.3427C17.69 28.4327 17.03 29.9127 17.03 31.5527C17.03 34.8627 19.72 37.5527 23.03 37.5527C25.82 37.5527 28.16 35.6327 28.83 33.0527H41.22C41.89 35.6327 44.23 37.5527 47.02 37.5527C50.33 37.5527 53.02 34.8627 53.02 31.5527C53.02 29.9127 52.36 28.4227 51.29 27.3427L57.32 17.3027C57.86 17.4627 58.43 17.5527 59.02 17.5527C61.81 17.5527 64.15 15.6327 64.82 13.0527H77.21ZM35.01 17.5527C37.8 17.5527 40.14 15.6327 40.81 13.0527H53.2C53.47 14.0927 54 15.0227 54.73 15.7627L48.7 25.8027C48.16 25.6427 47.59 25.5527 47 25.5527C44.21 25.5527 41.87 27.4727 41.2 30.0527H28.81C28.54 29.0127 28.01 28.0827 27.28 27.3427L33.31 17.3027C33.85 17.4627 34.42 17.5527 35.01 17.5527ZM38.14 25.0527C40 22.5227 42.87 20.8927 46 20.6027L47.53 18.0527H43.88C42.02 20.5827 39.15 22.2127 36.02 22.5027L34.49 25.0527H38.14Z"
          fill="currentColor"
        />
      </svg>
    </a>
  )
}

export default Logo
