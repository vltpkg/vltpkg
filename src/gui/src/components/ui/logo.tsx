import { type HTMLAttributes, forwardRef } from 'react'
import { cn } from '@/lib/utils.js'

export const Logo = forwardRef<
  HTMLParagraphElement,
  HTMLAttributes<HTMLHeadingElement>
>(({ className }, ref) => (
  <div
    ref={ref}
    className={cn(
      'flex flex-row items-center justify-center',
      className,
    )}>
    <h1 className="flex items-center justify-center">
      <span className="pr-2 text-2xl font-semibold leading-none tracking-wide">
        vlt
      </span>
      <span className="text-2xl font-light leading-none tracking-wide text-neutral-500">
        {'/v≈çlt/'}
      </span>
    </h1>
  </div>
))
Logo.displayName = 'Logo'
