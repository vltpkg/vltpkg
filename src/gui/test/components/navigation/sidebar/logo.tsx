import { vi, expect, afterEach, describe, it } from 'vitest'
import { render, cleanup } from '@testing-library/react'
import html from 'diffable-html'
import { useGraphStore as useStore } from '@/state/index.js'
import { SidebarLogo } from '@/components/navigation/sidebar/logo.jsx'

vi.mock('@/components/ui/sidebar.jsx', () => ({
  SidebarMenu: 'gui-sidebar-menu',
  SidebarMenuButton: 'gui-sidebar-menu-button',
  SidebarMenuItem: 'gui-sidebar-menu-item',
}))

vi.mock('@/components/icons/vlt-v.jsx', () => ({
  VLTV: 'gui-vlt-icon',
}))

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  const CleanUp = () => (useStore(state => state.reset)(), '')
  render(<CleanUp />)
  cleanup()
})

describe('AppSidebar Logo', () => {
  it('renders correctly', () => {
    const Container = () => {
      return <SidebarLogo />
    }

    const { container } = render(<Container />)
    expect(container.innerHTML).toMatchSnapshot()
  })
})
