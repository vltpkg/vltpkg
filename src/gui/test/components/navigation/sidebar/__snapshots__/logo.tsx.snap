// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppSidebar Logo > renders correctly 1`] = `

<gui-sidebar-menu>
  <gui-sidebar-menu-item>
    <gui-sidebar-menu-button
      size="lg"
      classname="whitespace-nowrap data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground"
      aschild="true"
    >
      <a href="/">
        <div class="flex aspect-square h-full items-center justify-center">
          <gui-vlt-icon
            color="white"
            classname="flex items-center justify-center"
          >
          </gui-vlt-icon>
        </div>
        <div class="grid flex-1 text-left text-sm leading-tight">
          <span class="truncate font-semibold">
            vlt
            <span class="ml-1 truncate font-light">
              /volt/
            </span>
          </span>
        </div>
      </a>
    </gui-sidebar-menu-button>
  </gui-sidebar-menu-item>
</gui-sidebar-menu>

`;
